src/components/dialogs/network-modifications/generator/modification/regulating-terminal-modification-dialog.js

```diff
@@ -0,0 +1,172 @@
+import { SelectInput } from '@gridsuite/commons-ui';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { gridItem } from 'components/dialogs/dialogUtils';
+import {
+    EQUIPMENT,
+    VOLTAGE_LEVEL,
+    VOLTAGE_REGULATION_TYPE,
+} from 'components/utils/field-constants';
+import yup from 'components/utils/yup-config';
+import PropTypes from 'prop-types';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormProvider, useForm, useWatch } from 'react-hook-form';
+import ModificationDialog from 'components/dialogs/commons/modificationDialog';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { REGULATION_TYPES } from 'components/network/constants';
+import { Grid } from '@mui/material';
+import RegulatingTerminalForm from 'components/dialogs/regulating-terminal/regulating-terminal-form';
+import { getTapChangerEquipmentSectionTypeValue } from 'components/utils/utils';
+import { fetchVoltageLevelsListInfos } from 'services/study/network';
+
+const RegulatingTerminalModificationDialog = ({
+    data,
+    currentNode,
+    studyUuid,
+    isUpdate,
+    editDataFetchStatus,
+    onModifyRegulatingTeminalGenerator,
+    ...dialogProps
+}) => {
+    const emptyFormData = useMemo(
+        () => ({
+            [VOLTAGE_REGULATION_TYPE]: null,
+            [VOLTAGE_LEVEL]: null,
+            [EQUIPMENT]: null,
+        }),
+        []
+    );
+
+    const formSchema = yup
+        .object()
+        .shape({
+            [VOLTAGE_REGULATION_TYPE]: yup.string().nullable().required(),
```

Comment:
in this case, the [VOLTAGE_REGULATION_TYPE] field can either have a string value or be null, but it cannot be any other type..
if it is not the required validation, you have to remove 'nullable()'